src/main/kotlin/Stores/ChannelSubscriptionStore.kt

```diff
@@ -0,0 +1,88 @@
+package Stores
+
+import models.ChannelSubscription
+import io.ktor.http.Parameters
+import models.Channel
+import models.User
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.eq
+import org.jetbrains.exposed.sql.insert
+import org.jetbrains.exposed.sql.select
+import org.jetbrains.exposed.sql.selectAll
+import org.joda.time.DateTime
+import java.util.*
+data class UsersSmallObj(
+        val username: String,
+        val id: String)
+
+data class ChannelUsers(
+        val name: String,
+        val id: String,
+        val User: UsersSmallObj
+)
+data class ChannelSubscriptionObj (
+        var userId: String,
+        var channelId: String,
+        var id: String,
+        var createdAt: String,
+        var updateAt: String
+)
+object ChannelSubscriptionStore {
+    suspend fun createChannelSubscription(params: Parameters): String? {
+        val uuid = UUID.randomUUID().toString()
+
+        DatabaseFactory.dbQuery {
+            ChannelSubscription.insert {
+                it[userId] = params["userId"]!!
+                it[channelId] = params["channelId"]!!
+
+                it[id] = uuid
+                it[createdAt] = DateTime.now()
+                it[updateAt] = DateTime.now()
+            }
+        }
+        return getChannelSubscription(uuid)
+    }
+    suspend private fun getChannelSubscription(uuid: String?): String? {
```

Comment:
we make the `uuid` nullable here, but we don't check and see if it's null inside of our function here, and we are returning a `String?` which is nullable. Maybe not, but just throwing it out there, but if the value we're passing in is null, we can go ahead and return an empty string, or we can leave it up to the caller of `getChannelSubscription()` to make the check for having to pass in String with a value. Does that make sense?

Reply:
you can use `try { call.parameters["uuid"]!! }` and return a response saying that the uuid is a required param, or something a bit more generic?

Reply:
though in this specific function i was checking if its null (uuid!!)

Reply:
`uuid!!` doesn't really check if it's null. it converts `uuid` to a non-null type, meaning that if it's null, we will get a NPE. This isn't true, but for the purposes of this explanation, `!!` is the only way to try and access a value in Kotlin that will throw a null-pointer exception. When we use `!!` we want to make sure that we actually do a check (uuid != null) at some point before calling `!!`. 

Take a gander at https://kotlinlang.org/docs/reference/null-safety.html for some null safety in Kotlin. Unless I've completely misinterpreted what you were saying. In which case, carry on